packages/server-wallet/src/wallet/index.ts

```diff
@@ -501,16 +506,22 @@ export class Wallet extends EventEmitter<WalletEvent>
 
     const wirePayload = validatePayload(rawPayload);
 
+    if (wirePayload.walletVersion !== WALLET_VERSION)
+      throw Error(
+        `Mismatch in wallet versions: this wallet is a ${WALLET_VERSION} but received a message from a ${wirePayload.walletVersion}`
+      );
```

Comment:
This is probably too restrictive for the moment e.g. if we want to make a performance improvement for the wallet deployed in the gateway, then this would break all the indexers. Maybe a warning - but even that could become pretty spammy, as we'd be logging a line for every single state both on the gateway and the indexer.

It seems like the time we really need to know this information is if something goes wrong. Ideally we'd somehow add this information to any error thrown or logged during the execution of a `pushMessage` - something like `{ ourWalletVersion: ..., theirWalletVersion: ...}`. We might be able to do that with a `try / catch` block around the `pushMessage`.

Reply:
d170b0a, 140a77c

Reply:
You can log the first time you see it, so it's not spammy

Reply:
What I settled on is adding versioning information both to i) existing logs and ii) existing errors. I hope that qualifies it as "not spammy"?